kinesis/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/sinks/KinesisSink.scala

```diff
@@ -206,6 +209,38 @@ object KinesisSink {
       case None => None.asRight
     }
 
+  /**
+    * Check whether an SQS queue (buffer) exists
+    * @param name Name of the queue
+    * @return Whether the queue exists
+    */
+  private def sqsQueueExists(client: AmazonSQS, name: String): Boolean =
+    try {
+      client.getQueueUrl(name)
+      true
+    } catch {
+      case _: QueueDoesNotExistException => false
```

Comment:
There are no other types of exceptions to handle, e.g. timeout or network issues? I wonder if we should add a `case NonFatal(e)`